### DISCUSSION TITLE
Suggestions list

### DISCUSSION THREAD

**Discussion by mradenovic** [SimplQ/simplQ-frontend#495] [Ideas]
It's a lot of ideas. Before I clutter the tracker with a ton of new issues, let me know what you think.

**Use Github Discussions for open discussions**

Any of the following ideas will probably need a new issue and discussion, and all comments and critics are welcome 😄. But every discussion does not need an issue. I have never used [Github Dicussions](https://docs.github.com/en/discussions/quickstart) and it might be worth trying.

**Use Project boards to organize and prioritize issues**

Some of the issues have to be resolved before others. It would be helpful to organize everything in a Kanban view. I have used [Project boards](https://docs.github.com/en/github/managing-your-work-on-github/about-project-boards) before, and it was useful even on one-man projects.

**Change naming convention**

Currently `index.jsx` files contain component code. It is a little bit tricky when one opens three tabs with different classes inside and tries to navigate thru code. I suggest the following:
- All filenames except `index.jsx` should be in `PascalCase` (i.e. `Join.jsx`, `Join.module.scss`, `Form.jsx`)
- Use `index.jsx` to re-export classes from the folder

**Redesign page flow**

As discussed [here](https://github.com/SimplQ/simplQ-frontend/pull/482#discussion_r563594576), Page flow doesn't look great on all screen sizes and needs a new issue. It can be done very simple with [flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/), or more complex with [grid](https://css-tricks.com/snippets/css/complete-guide-grid/). The decision depends on how complex the app will be, which brings us to the next topic, **Features** 😄 

**Describe desired features**

> There are a ton of features being planned. So if you are considering contributing to this repository, please first discuss the change you wish to make via the issue tracker.

[Contributing](https://github.com/SimplQ/simplQ-frontend#contributing) section should have a list, or a link to a document, with planned features.

**Describe deployment process**

Currently, it looks like there is only one server deployed. It is used for production and development. There should be a **How to** section to describe how to deploy your own server (both for backend and frontend). At the end of the day, if one just wants to organize a queue for a temp lemon stand in front of the garage, the current state will do 😄. For any serious deployment, one would like to know how to do it at home.

**i18n**

All of this already looks great, and it can be used as-is if one speaks English. I would love to see it in other languages.

**Comment by daltonfury42** [comment]
Hi @mradenovic Thanks a lot of this. There is a lot of great feedback in here.

**Use Github Discussions for open discussions:** I have never used this feature before, but looking at the docs, I feel we could find good uses for it. I will enable it, try to set up some topics, add some links to it from readme and lets see how it goes. This would take some time though. 

**Use Project boards to organise and prioritise issues:** @maaverik and I develop and maintain SimplQ in our free time, and I don't see a need for a Kanban board right now. We did try using Project Boards during initial development phase, but it got put on the back burner. At our scale, what had worked well for us was to break down a page/feature into small independent issues. While we work on priority issues or ones that require creating the basic skeleton, happy to find a lot of outside collaborators chipping in with PRs for remaining/follow up tasks.

**Redesign page flow** and **Change naming convention**: Agreed, we should do this.

**Describe deployment process**: We have instructions to run backend locally [here](https://github.com/SimplQ/simplQ-frontend/issues/487). But I get what you mean. I was thinking of several options; one possibility is to create a docker image and push to docker hub. What do you think?

**i18n:** Yes, internationalisation is in the roadmap, we have not started working on it yet. 

**Describe desired features**: I have a couple of things in mind, yes, I need to write them down. Will take this up, and I could also try Github Discussions for this.

In an effort to win real business clients, sending SMS alerts to users who join the queue is what I have as my next priority item. (discussed [here](https://github.com/SimplQ/simplQ-frontend/issues/257)). But since this is a side gig, things progress slowly over weekends. If you would like to drive any of the discussed items, please go ahead, I am more than happy to collaborate. 

 @maaverik what do you think?

**Comment by maaverik** [comment]
Agreed, these are great ideas @mradenovic. It might be slow, but we'll be able to achieve them I'm sure.

**Comment by mradenovic** [comment]
> **Redesign page flow** and **Change naming convention**: Agreed, we should do this.

This is where I could jump in. I feel like renaming would be the first thing to do, but I don't want to cause issues with a big change,

> **Describe deployment process**: We have instructions to run backend locally [here](https://github.com/SimplQ/simplQ-frontend/issues/487). But I get what you mean. I was thinking of several options; one possibility is to create a docker image and push to docker hub. What do you think?

The link is wrong, did you mean [backend dev setup](https://github.com/SimplQ/simplQ-backend#development-envirmonment-setup-instructions)? Docker or deployment script would be great.

> **i18n:** Yes, internationalisation is in the roadmap, we have not started working on it yet.

I'll look into this and open an issue for that.

> In an effort to win real business clients, sending SMS alerts to users who join the queue is what I have as my next priority item. (discussed [here](https://github.com/SimplQ/simplQ-frontend/issues/257)). But since this is a side gig, things progress slowly over weekends. If you would like to drive any of the discussed items, please go ahead, I am more than happy to collaborate.

👍 for SMS alerts 😄

**Comment by daltonfury42** [comment]
I raised https://github.com/SimplQ/simplQ-frontend/issues/489 for self deployment process.

If you are going to tackle redesigning the page flow, that would be really helpful. I went ahead and created an issue and added my thoughts [here](https://github.com/SimplQ/simplQ-frontend/issues/490). Let me know what you think, if you have any different ideas or need more information, let us discuss there.

Can you create an issue for the naming conventions as well? For non-urgent issues that I think can be taken up by a first time contributor, I usually define the issue and move on.
